### PR TITLE
[FLINK-2956] [tests] Migrate integration tests for Table API.

### PR DESCRIPTION
**Pull request by gallenvara**
Migrate integration tests of Table API from temp file to `collect()` as described in umbrella jira.

### CODE REVIEW

**Comment by chiwanpark** [comment]
I would like to shepherd this PR. Thanks for opening pull request @gallenvara. I'll review soon.

**Comment by chiwanpark** [comment]
Hi @gallenvara, I read your PR and have some comments.

First, since Flink 0.10, we dropped Java 6 support and can use omit types if the types can be inferred. So could you remove type description in that case? With removing, we can make codes more readable.

Second, to convert Scala sequence to Java list, using `JavaConverters` is better than `JavaConversions`. For example:

```scala
import scala.collection.JavaConverters._

~skipped~

val result = ds.collect()
val expected = "~~~"

TestBaseUtils.compareResultAsText(result.asJava, expected)
```

After importing `scala.collection.JavaConverters._`, we can convert Scala collection to Java collection with `asJava` method.

**Comment by gallenvara** [comment]
Hi, @chiwanpark . Thanks for your review. I'm not very clear about your first suggestion. Do you mean that we should remove type description when to create a new generics? 
For example: 
`DataSource<Tuple2<Float, String>> input = env.fromElements(new Tuple2<Float, String>(1f, "Hello"))`
-->
`DataSource<Tuple2<Float, String>> input = env.fromElements(new Tuple2<>(1f, "Hello"))`

**Comment by chiwanpark** [comment]
@gallenvara Yes, exactly right.

**Comment by gallenvara** [comment]
Hi, @chiwanpark . I have modified the code and committed a new commit. Can you help me with review work?

**Comment by gallenvara** [comment]
Hi, @chiwanpark . I have modified the code and submitted a new commit. Can you help me with review work?

**Comment by chiwanpark** [comment]
Okay I'll review soon. Thanks for update.

File: flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/AsITCase.java [hunk 1]
```diff
@@ -68,16 +49,14 @@ public void testAs() throws Exception {
 				tableEnv.fromDataSet(CollectionDataSets.get3TupleDataSet(env), "a, b, c");
 
 		DataSet<Row> ds = tableEnv.toDataSet(table, Row.class);
-		ds.writeAsText(resultPath, FileSystem.WriteMode.OVERWRITE);
-
-		env.execute();
-
-		expected = "1,1,Hi\n" + "2,2,Hello\n" + "3,2,Hello world\n" + "4,3,Hello world, " +
+		List<Row> results =ds.collect();
```
Comment (chiwanpark) [inline flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/AsITCase.java]: Please add a space after `=`.

File: flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/CastingITCase.java [hunk 2]
```diff
@@ -44,29 +41,13 @@ public CastingITCase(TestExecutionMode mode){
...
-
 	@Test
 	public void testAutoCastToString() throws Exception {
 		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
 		TableEnvironment tableEnv = new TableEnvironment();
 
 		DataSource<Tuple7<Byte, Short, Integer, Long, Float, Double, String>> input =
-				env.fromElements(new Tuple7<Byte, Short, Integer, Long, Float, Double, String>(
+				env.fromElements(new Tuple7<>(
 						(byte) 1, (short) 1, 1, 1L, 1.0f, 1.0d, "Hello"));
```
Comment (chiwanpark) [inline flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/CastingITCase.java]: I think merging L51 and L50 would be better. After omitting generic parameters, we don't need a newline for this.

File: flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/CastingITCase.java [hunk 3]
```diff
@@ -89,7 +68,7 @@ public void testNumericAutocastInArithmetic() throws Exception {
...
```
Comment (chiwanpark) [inline flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/CastingITCase.java]: Same as above. Please remove newline starting this.

File: flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/ExpressionsITCase.java [hunk 4]
```diff
@@ -98,11 +77,9 @@ public void testLogic() throws Exception {
 				"b && true, b && false, b || false, !b");
 
 		DataSet<Row> ds = tableEnv.toDataSet(result, Row.class);
-		ds.writeAsText(resultPath, FileSystem.WriteMode.OVERWRITE);
-
-		env.execute();
-
-		expected = "true,false,true,false";
+		List<Row> results =ds.collect();
```
Comment (chiwanpark) [inline flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/ExpressionsITCase.java]: Please add a space after `=`.

File: flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/FilterITCase.java [hunk 5]
```diff
@@ -73,11 +54,9 @@ public void testAllRejectingFilter() throws Exception {
 				.filter("false");
 
 		DataSet<Row> ds = tableEnv.toDataSet(result, Row.class);
-		ds.writeAsText(resultPath, FileSystem.WriteMode.OVERWRITE);
-
-		env.execute();
-
-		expected = "\n";
+		List<Row> resutls = ds.collect();
```
Comment (chiwanpark) [inline flink-staging/flink-table/src/test/java/org/apache/flink/api/java/table/test/FilterITCase.java]: Typo in variable name: `resutls` -> `results`

**Comment by chiwanpark** [comment]
@gallenvara Sorry for late. I reviewed your changes. Except some minor comments, looks good to merge for me.

**Comment by gallenvara** [comment]
Hi, @chiwanpark . I have modified the code format and submitted a new commit.

**Comment by chiwanpark** [comment]
@gallenvara Thanks for update! Looks good to merge for me. If there is no objection until tomorrow, I'll merge this.

**Comment by aljoscha** [comment]
Looks good, please go ahead @chiwanpark :+1:

**Comment by chiwanpark** [comment]
Merging...

**Comment by chiwanpark** [comment]
@gallenvara Because I forgot add close message in the commit, the PR should be closed manually. Could you close this PR? This change is merged. Sorry for inconvenience.

**Pull request closed without merging**